packages/venia-ui/lib/components/AuthModal/authModal.js

```diff
@@ -1,51 +1,27 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React from 'react';
 import { func, shape, string } from 'prop-types';
-import { useUserContext } from '@magento/peregrine/lib/context/user';
 
 import { mergeClasses } from '../../classify';
 import CreateAccount from '../CreateAccount';
 import ForgotPassword from '../ForgotPassword';
 import MyAccount from '../MyAccount';
 import SignIn from '../SignIn';
 import defaultClasses from './authModal.css';
-
-const UNAUTHED_ONLY = ['CREATE_ACCOUNT', 'FORGOT_PASSWORD', 'SIGN_IN'];
+import { useAuthModal } from '@magento/peregrine/lib/mixins/AuthModal/useAuthModal';
 
 const AuthModal = props => {
-    const {
-        closeDrawer,
-        showCreateAccount,
-        showForgotPassword,
-        showMainMenu,
-        showMyAccount,
-        view
-    } = props;
-
-    const [username, setUsername] = useState('');
-    const [userState, { createAccount, signOut }] = useUserContext();
-    const { currentUser } = userState;
     const classes = mergeClasses(defaultClasses, props.classes);
     let child = null;
 
-    const resetDrawer = useCallback(() => {
-        showMainMenu();
-        closeDrawer();
-    }, [closeDrawer, showMainMenu]);
-
-    // if the user is authed, the only valid view is "MY_ACCOUNT"
-    useEffect(() => {
-        if (currentUser && currentUser.id && UNAUTHED_ONLY.includes(view)) {
-            showMyAccount();
-        }
-    }, [currentUser, showMyAccount, view]);
-
-    const handleCreateAccount = useCallback(
-        async values => {
-            await createAccount(values);
-            showMyAccount();
-        },
-        [createAccount, showMyAccount]
-    );
+    const {
+        handleClose,
+        handleCreateAccount,
+        handleSignOut,
```

Comment:
It strikes me suddenly that these may also be areas where developers may want to customize behaviors. Ex: send an analytic event, pop up a dialog, etc. in addition to the default behavior.

Totally out of scope for this PR but I'd be interested in any thoughts y'all have had on this front.

Reply:
I love the consistency of the API coming back from the hook 🥇